packages/ketcher-polymer-editor-react/src/components/shared/ui/dialog/Modal.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  useTheme
+} from '@mui/material'
+import React, { useMemo, useCallback } from 'react'
+import { Icon } from 'components/shared/ui/icon'
+import styled from '@emotion/styled'
+
+import { scrollbarThin } from 'styles/mixins'
+
+interface ModalProps {
+  children: JSX.Element | Array<JSX.Element>
+  title: string
+  isModalOpen: boolean
+  setIsModalOpen: (isOpen: boolean) => void
+}
+
+const ModalHeader = styled(DialogTitle)(({ theme }) => ({
+  margin: '12px 11px 12px 16px',
+  padding: 0,
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  fontFamily: `${theme.font.family.inter}`,
+  fontSize: `${theme.font.size.medium}`,
+  fontWeight: `${theme.font.weight.regular}`,
+  textTransform: 'uppercase'
+}))
+
+const Title = styled.div({
+  marginRight: '10px'
+})
+
+const Content = styled(DialogContent)(({ theme }) => ({
+  margin: '0 16px 16px',
+  padding: 0,
+
+  ...scrollbarThin(theme)
+}))
+
+const Footer = styled(DialogActions)({
+  margin: '0 16px 16px',
+  padding: 0
+})
+
+type ModalSubcomponent = 'Content' | 'Footer'
+
+export const Modal = ({
+  title,
+  isModalOpen,
+  setIsModalOpen,
+  children
+}: ModalProps) => {
```

Comment:
I think Modal should also accept an `onOk` handler, because we will probably use it to confirm some actions.